climpred_skeleton/time.py

```diff
@@ -0,0 +1,89 @@
+import warnings
+from typing import Union
+
+import cftime
+import pandas as pd
+import xarray as xr
+
+from climpred_skeleton.core import Verification
+
+
+class TimeManager(Verification):
+    def __init__(self, initialized, observation):
+        super().__init__(initialized, observation)
+        # Convert `init` and `time` indices to CFTimeIndex.
+        self._convert_to_cftime_index()
+        self._units = self._initialized['lead'].attrs['units']
+
+    def _convert_to_cftime_index(self, calendar: str = 'DatetimeProlepticGregorian'):
+        """Converts time indices for the prediction and observations to a
+        CFTimeIndex."""
+
+        def _return_converted_time_index(
+            time_index: Union[
+                xr.CFTimeIndex, pd.DatetimeIndex, pd.Float64Index, pd.Int64Index
+            ]
+        ):
+            assume_annual = False
+
+            if not isinstance(time_index, xr.CFTimeIndex):
```

Comment:
Is this extra nesting necessary?
What if it's just
```
if isinstance(time_index, pd.DateTimeIndex):
...
elif isinstance(time_index, pd.Float64Index)
...
elif not isinstance(time_index, xr.CFTimeIndex):
   ~~raise valueerror
```

Reply:
I think it's necessary since they all run the following under the main if statement

```python

cftime_dates = [
                    getattr(cftime, calendar)(int(y), int(m), int(d))
                    for (y, m, d) in split_dates
                ]
                time_index = xr.CFTimeIndex(cftime_dates)
```